lms/static/scripts/frontend_apps/components/StudentSelector.js

```diff
@@ -0,0 +1,95 @@
+import { createElement } from 'preact';
+import propTypes from 'prop-types';
+
+/**
+ * A student navigation tool which shows a student selection list and a previous and next button.
+ */
+
+export default function StudentSelector({
+  onSelectStudent,
+  selectedStudentIndex,
+  students,
+}) {
+  // Disable the next button if at the end of the list. The length is equal to
```

Comment:
I like the current behavior of being able to page through all students as well as being able to click back into "All Students." These comments here are helpful. Looking at the slightly fiddly arithmetic (which warrants the comments), I'm wondering if the logic here isn't about having next or previous _students_, but having next or previous _views_. i.e. this is more about paging forward and back overall. Something as simple as renaming these values might bring a little more clarity, e.g. `hasNextView`, `hasPrevView`. Or maybe that's _too_ generic. I'll leave it up to you!

Reply:
I like this, since the first slot is not not really a student, this makes more sense.